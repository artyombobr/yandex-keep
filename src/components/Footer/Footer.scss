.footer {
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 54px;
  background: white;
  margin-top: 32px;
  padding: 0 14px;
  font-size: 16px;
  line-height: 32px;

  &__null {
    display: none;
    opacity: 0;
  }

  &__name {
    flex-grow: 1;
  }

  &__copyright {
    flex-grow: 1;
    text-align: right;
  }
}

@media (min-width: 768px) {
  .footer {
    height: 64px;

    &__null {
      display: block;
    }

    &__name {
      text-align: right;
    }
  }
}
