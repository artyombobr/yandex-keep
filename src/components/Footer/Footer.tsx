import React from 'react';
import './Footer.scss';

const Footer = () => (
  <footer className='footer'>
    <div className='footer__null'>Яндекс ШРИ</div>
    <p className='footer__name'>Артём Бобр</p>
    <p className='footer__copyright'>Яндекс ШРИ</p>
  </footer>
);

export default Footer;
