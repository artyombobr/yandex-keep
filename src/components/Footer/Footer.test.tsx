import React from 'react';
import { shallow } from 'enzyme';
import Footer from './Footer';

describe('Footer', () => {
  it('renders correctly', () => {
    const component = shallow(<Footer />);
    expect(component).toMatchSnapshot();
  });
});
