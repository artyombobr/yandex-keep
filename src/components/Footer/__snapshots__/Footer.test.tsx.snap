// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="footer__null"
        >
          Яндекс ШРИ
        </div>,
        <p
          className="footer__name"
        >
          Артём Бобр
        </p>,
        <p
          className="footer__copyright"
        >
          Яндекс ШРИ
        </p>,
      ],
      "className": "footer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Яндекс ШРИ",
          "className": "footer__null",
        },
        "ref": null,
        "rendered": "Яндекс ШРИ",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Артём Бобр",
          "className": "footer__name",
        },
        "ref": null,
        "rendered": "Артём Бобр",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Яндекс ШРИ",
          "className": "footer__copyright",
        },
        "ref": null,
        "rendered": "Яндекс ШРИ",
        "type": "p",
      },
    ],
    "type": "footer",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="footer__null"
          >
            Яндекс ШРИ
          </div>,
          <p
            className="footer__name"
          >
            Артём Бобр
          </p>,
          <p
            className="footer__copyright"
          >
            Яндекс ШРИ
          </p>,
        ],
        "className": "footer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Яндекс ШРИ",
            "className": "footer__null",
          },
          "ref": null,
          "rendered": "Яндекс ШРИ",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Артём Бобр",
            "className": "footer__name",
          },
          "ref": null,
          "rendered": "Артём Бобр",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Яндекс ШРИ",
            "className": "footer__copyright",
          },
          "ref": null,
          "rendered": "Яндекс ШРИ",
          "type": "p",
        },
      ],
      "type": "footer",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
